connect/config.go

```diff
@@ -0,0 +1,99 @@
+package connect
+
+import (
+	"fmt"
+	"net"
+	"net/url"
+	"regexp"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type Config struct {
+	host string
+	port string
+	// requireTLS is whether TLS should be used or cleartext
+	requireTLS bool
+	// opts are additional options to pass to the gRPC client
+	opts []grpc.DialOption
+}
+
+// NewConfig returns a new Config from an endpoint string, that has to be in a URL format.
+// The endpoint can be either http:// or https:// that will be used to determine whether TLS should be used.
+// if port is not specified, it will be inferred from the scheme (80 for http, 443 for https).
+func NewConfig(endpoint string) (*Config, error) {
+	c := new(Config)
+	err := c.parseEndpoint(endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("invalid endpoint: %w", err)
+	}
+	return c, nil
+}
+
+// GetConnectionURI returns connection string conforming to https://github.com/grpc/grpc/blob/master/doc/naming.md
+func (c *Config) GetConnectionURI() string {
+	return fmt.Sprintf("dns:%s:%s", c.host, c.port)
```

Comment:
(optional)

If this is the only place that `host` and `port` are used, would it make sense to combine these fields into a single `connectionURI` field, set within `parseEndpoint()`?